utils/utils.py

```diff
@@ -0,0 +1,299 @@
+
+  def create_lattice(self, iexpr: pynini.FstLike,
+                     oexpr: pynini.FstLike) -> pynini.Fst:
+    """Creates edit lattice for a pair of input/output strings or acceptors.
+    Args:
+      iexpr: input string or acceptor
+      oexpr: output string or acceptor.
+    Returns:
+      A lattice FST.
+    """
+    sybt = dict()
+    syn_list = load_labels(get_abs_path("synonym.tsv"))
+    for wordlist in syn_list:
+      for word in wordlist:
+        if word in iexpr:# syn found
+          if len(word.split(" "))>1:#syn with more than one word's
+            wordstring = word.split(" ")[0]
+            for j in range(1,len(word.split(" "))):
+              wordstring+="_"+word.split(" ")[j]
+            tagger = pynini.cross(word,wordstring)
+            tagger = pynini.cdrewrite(tagger,"","",pynini.closure(utf8.VALID_UTF8_CHAR))        
+            iexpr = (iexpr @ tagger).string()
+    iex_list = iexpr.split(" ")
+    oex_list = oexpr.split(" ")
+    iex = pynini.accep("")
+    oex = pynini.accep("")
+    lastword = None
+    for word in iex_list:
+      flag = True
+      if "_" in word:# show more than one word cases:
+        word_more_one = word.split("_")
+        word_more_one_string = word_more_one[0]
+        for j in range(1,len(word_more_one)):
+          word_more_one_string+=" " + word_more_one[j]
+          for wordlist in syn_list:
+            if word_more_one_string in wordlist:
+              temp = pynini.accep("")
+              for synword in wordlist:
+                if len(synword.split(" "))>1:
+                  ttemp = pynini.accep("")
+                  synword_s = synword.split(" ")
+                  for sword in synword_s:
+                    idstate = pynini.accep('['+ sword + ']').paths().ilabels()[0]
+                    sybt[idstate] = sword
+                    ttemp+=pynini.accep('['+ sword + ']')
+                  temp|=ttemp
+                else:
+                  idstate = pynini.accep('['+ synword + ']').paths().ilabels()[0]
+                  sybt[idstate] = synword
+                  temp|=pynini.accep('['+ synword + ']')
+              iex+=temp
+      else:
+        for wordlist in syn_list: 
+          if word in wordlist:#替换词，直接索引dict
+            temp = pynini.accep("")
+            for synword in wordlist:
+                if len(synword.split(" "))>1:
+                  ttemp = pynini.accep("")
+                  synword_s = synword.split(" ")
+                  for sword in synword_s:
+                    idstate = pynini.accep('['+ sword + ']').paths().ilabels()[0]
+                    sybt[idstate] = sword
+                    ttemp+=pynini.accep('['+ sword + ']')
+                  temp|=ttemp
+                else:
+                  idstate = pynini.accep('['+ sword + ']').paths().ilabels()[0]
+                  sybt[idstate] = sword
+                  temp|=pynini.accep('['+ synword + ']')
+            iex+=temp
+            flag = False
```

Comment:
I haven't go through this code carefully, major advice is that these iexpr & oexpr graph construction code should be moved outside, e.g. to asr-score_v2.py.

Please cleanup this code and variable names a little bit, if you can.